### PR TITLE
Avoid panic and agent failure in case of not DSN or ApiKey

### PR DESCRIPTION
**Pull request by tonyredondo**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -193,30 +193,36 @@ func NewAgent(options ...Option) (*Agent, error) {
...
+		} else {
+			agent.logger.Println("SCOPE_DSN can't be found in the env vars.")
 		}
 	}
 
 	if agent.apiKey == "" {
 		if apikey, set := os.LookupEnv("SCOPE_APIKEY"); set && apikey != "" {
 			agent.apiKey = apikey
 		} else if configProfile != nil {
+			agent.logger.Println("Api key found in the native app configuration.")
```
Comment (fermayo) [inline agent/agent.go]:
```suggestion
			agent.logger.Println("API key found in the native app configuration")
```

File: agent/agent.go [hunk 2]
```diff
@@ -193,30 +193,36 @@ func NewAgent(options ...Option) (*Agent, error) {
...
 			dsnApiKey, dsnApiEndpoint, dsnErr := parseDSN(dsn)
 			if dsnErr != nil {
-				agent.logger.Printf("Error parsing dsn value: %v", dsnErr)
+				agent.logger.Printf("Error parsing dsn value: %v\n", dsnErr)
 			} else {
 				agent.apiKey = dsnApiKey
 				agent.apiEndpoint = dsnApiEndpoint
 			}
+		} else {
+			agent.logger.Println("SCOPE_DSN can't be found in the env vars.")
```
Comment (fermayo) [inline agent/agent.go]:
```suggestion
			agent.logger.Println("environment variable $SCOPE_DSN not found")
```

File: agent/agent.go [hunk 3]
```diff
@@ -193,30 +193,36 @@ func NewAgent(options ...Option) (*Agent, error) {
...
 
 	if agent.apiKey == "" {
 		if apikey, set := os.LookupEnv("SCOPE_APIKEY"); set && apikey != "" {
 			agent.apiKey = apikey
 		} else if configProfile != nil {
+			agent.logger.Println("Api key found in the native app configuration.")
 			agent.apiKey = configProfile.ApiKey
 		} else {
-			return nil, errors.New("Scope API key could not be autodetected")
+			agent.logger.Println("Api key not found, agent can't be started.")
```
Comment (fermayo) [inline agent/agent.go]:
```suggestion
			agent.logger.Println("API key not found, agent can't be started")
```

File: agent/agent.go [hunk 4]
```diff
@@ -193,30 +193,36 @@ func NewAgent(options ...Option) (*Agent, error) {
...
+			return nil, errors.New("scope API key could not be autodetected")
 		}
 	}
 
 	if agent.apiEndpoint == "" {
 		if endpoint, set := os.LookupEnv("SCOPE_API_ENDPOINT"); set && endpoint != "" {
 			agent.apiEndpoint = endpoint
 		} else if configProfile != nil {
+			agent.logger.Println("Api endpoint found in the native app configuration.")
```
Comment (fermayo) [inline agent/agent.go]:
```suggestion
			agent.logger.Println("API endpoint found in the native app configuration")
```

File: agent/agent.go [hunk 5]
```diff
@@ -193,30 +193,36 @@ func NewAgent(options ...Option) (*Agent, error) {
...
 			agent.apiEndpoint = configProfile.ApiEndpoint
 		} else {
+			agent.logger.Printf("Using default endpoint: %v.\n", defaultApiEndpoint)
```
Comment (fermayo) [inline agent/agent.go]:
```suggestion
			agent.logger.Printf("using default endpoint: %v\n", defaultApiEndpoint)
```

**Comment by tonyredondo** [comment]
Added a message when Scope fails to initialize, please check @fermayo 

image

**Pull request merged**